tests/runtests.q

```diff
@@ -1,8 +1,42 @@
-KUltd each hsym`$.proc.params[`test];
+// Set up results and logging directories
+.k4.setup:{[respath;testname]
+  if[not 11h=type key hsym `$respath;system "mkdir -p ",respath];
+  if[not 11h=type key hsym `$logpath:raze respath,"/logs/";system "mkdir ",logpath];
+  .proc.createlog[logpath;testname;`$ssr[string .z.p;"[D.:]";"_"];0b];
+  };
+
+// Write test results to disk
+.k4.writeres:{[res;err;respath;rtime;testname]
+  // Generate file names and create them if they don't exist
+  resfile:"results_",(raze "." vs string .z.d),".csv";
+  errfile:"failures_",(raze "." vs string .z.d),".csv";
+  if[not 11h=type key hsym `$respath;system "mkdir -p ",respath];
+  if[not -11h=type key hsym `$rf:respath,"/",resfile;system "touch ",rf;.lg.o[testname;"Creating file ",rf]];
+  if[not -11h=type key hsym `$ef:respath,"/",errfile;system "touch ",ef;.lg.o[testname;"Creating file ",ef]];
+
+  // Timestamp tests
+  res:`runtime xcols update runtime:first rtime from delete timestamp from res;
+  err:`runtime xcols update runtime:first rtime from delete timestamp from err;
+
+  // Write test results and errors to these files
+  .lg.o[testname;"Writing ",string[count KUTR]," results rows and ",string[count KUerr]," error rows"];
+  hclose abs neg[hopen hsym `$rf] $[hcount hsym `$rf;1;0]_csv 0: res;
+  hclose abs neg[hopen hsym `$ef] $[hcount hsym `$ef;1;0]_csv 0: err;
```

Comment:
```suggestion
  hclose abs neg[hopen rf] $[hcount rf:hsym `$rf;1;0]_csv 0: res;
  hclose abs neg[hopen ef] $[hcount ef:hsym `$ef;1;0]_csv 0: err;
```

think these lines could do with a little explanation, there's a number of things happening & it's not all immediately clear why IMO